lib/primitives/mtx.js

```diff
@@ -1635,7 +1638,7 @@ class CoinSelector {
     if (this.tx.view.hasEntry(coin))
       return false;
 
-    if (coin.covenant.isNonspendable())
+    if (coin.covenant.isNonspendable() || (coin.covenant.isOpen() && this.selection === 'sweepdust'))
```

Comment:
There are also a lot of 0 value redeems to skip etc. I think we should reorganize this as follows:

```
if (this.selection === 'sweepdust' && VALUE === 0)
  return false;
```